### PR TITLE
make sure all commands return a dict. Fix #5224

### LINKED ISSUE #5224: all commands must return a dict containing `commandStatus` key
This is needed so that `crab.py` can know if a command succeeded of failed and properly set the exit code.
Currently commands always exit with 0 and rely on user reading the messages to tell if things worked.
This prevents constructing a decent testing.

Progress tracker

- [x] checkdataset
- [x]  checkusername
- [x]  checkwrite
- [x]  createmyproxy
- [x]  getlog
- [x]  getoutput
- [x]  kill
- [x]  preparelocal
- [x]   proceed
- [x]   remake
- [x]   report
- [x]   resubmit
- [x]  status
- [x]   submit
- [x]   tasks
- [x]   uploadlog

### PR DESCRIPTION
**Pull request by belforte**

### CODE REVIEW

**Comment by belforte** [comment]
@mapellidario note that this assumes that https://github.com/dmwm/CRABClient/issues/5224#issuecomment-1765315793 is OK. I am still waiting yours and Wa's opinion on both the strategy and the  implementation.

**Review by mapellidario — APPROVED** [review]
I like it, thanks! 
I do not expect many users to use these return values, so i think it is pretty safe to change them and just announce the changes

File: src/python/CRABClient/Commands/remake.py [hunk 1]
```diff
@@ -36,8 +37,15 @@ def remakecache(self,taskname):
...
-            return requestarea
 
+            caller = sys._getframe(1)
+            if not 'crab.py' in caller.f_code.co_filename:
+                msg = "ATTENTION return value for 'remake' has been changed to a dictionary"
+                msg += "\n format is {'commandStatus': 'SUCCESS' or 'FAILED',"
+                msg += "\n            'workDir': name of the work directory created}"
+                self.logger.warning(msg)
+            returnDict = {'commandStatus': 'SUCCESS', 'workDir': requestarea}
+            return returnDict
```
Comment (novicecpp) [inline src/python/CRABClient/Commands/remake.py:48]: No need of warning message if you do not support backward compatibility.

File: bin/crab.py [hunk 2]
```diff
@@ -142,7 +142,9 @@ def log_exception(exc_type, exc_value, tback):
             sys.exit(-1)
         self.cmd = sub_cmd(self.logger, args[1:])
 
-        self.cmd()
+        returnDict = self.cmd()
+        if returnDict['commandStatus'] != 'SUCCESS':
+            sys.exit(1)
```
Comment (novicecpp) [inline bin/crab.py:147]: We should return/raise back to `__main__` section instead of exit here.

**Review by novicecpp — COMMENTED** [review]
I have some inline comments and also we should put some docstring about return value of subcommand (maybe in both SubCommand.py and RawCommand.py).

File: src/python/CRABClient/Commands/remake.py (hunk 1 above)
Comment (belforte) [reply src/python/CRABClient/Commands/remake.py:48]: I hope that a warning + a failure is more clear than just an obscure failure on some code which used to work. It surely makes no sense long term, but simply to help the transition. I do not think I have a way to catch if user code fails on my new return value or not.

File: bin/crab.py (hunk 2 above)
Comment (belforte) [reply bin/crab.py:147]: I agree. I put `sys.exit(1)` becasue raising ClientException did not seem adequate. In a way what I want is to make them end with exit code 1 some commands that until now reports a "failure" in the messages but return with exit code 0.
I will define a new exception for this

**Comment by novicecpp** [comment]
Yes.
Thanks Stefano.

**Comment by belforte** [comment]
all those pylint errors are due to "old code which needs to run in python2 as well, did not follow naming conventions and is anyhow untouched and working'

**Pull request merged**